### ISSUE TITLE
CUDNN_STATUS_EXECUTION_FAILED

### ISSUE THREAD

**Issue by DarshanDeshpande** [tensorflow/tensorflow#33938]
**System information**
- I have written custom code (as opposed to using a stock example script provided in TensorFlow): Slightly modified version of seq2seq model from keras documentation 
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows 10
- TensorFlow installed from (source or binary): Binary
- TensorFlow version (use command below): 2.0.0 Stable
- Python version: 3.7.5
- CUDA/cuDNN version: CUDA-10.0, CuDNN-7.6.4.38
- GPU model and memory: Nvidia 1050Ti(4GB) + 16GB RAM

**Describe the current behavior**
I'm getting an error while training on a small dataset using a seq2seq model. It trains upto 4 or 5 epochs after which it throws an error saying "CUDNN_STATUS_EXECUTION_FAILED" along with "Failed to call ThenRnnBackward"

**Describe the expected behavior**
I have tried reducing batch size and LSTM latent dimension but it crashes irrespectively. I recently updated to tf 2.0.0 and I never faced any such issues on 1.x

**Code to reproduce the issue**

batch_size = 16
epochs = 50
latent_dim = 50
training_samples = 5000

encoder_input_data = np.zeros((len(input_texts), max_encoder_seq_length, num_encoder_tokens),dtype='float32')
decoder_input_data = np.zeros((len(input_texts), max_decoder_seq_length, num_decoder_tokens),dtype='float32')
decoder_target_data = np.zeros((len(input_texts), max_decoder_seq_length, num_decoder_tokens),dtype='float32')

for i, (input_text, target_text) in enumerate(zip(input_texts, output_texts)):
    for t, char in enumerate(input_text):
        encoder_input_data[i, t, input_token_index.get(char)] = 1.
    encoder_input_data[i, t + 1:, input_token_index.get(' ')] = 1.

    for t, char in enumerate(target_text):
        decoder_input_data[i, t, target_token_index.get(char)] = 1.
        if t > 0:
            decoder_target_data[i, t - 1, target_token_index.get(char)] = 1.
    decoder_input_data[i, t + 1:, target_token_index.get(' ')] = 1.
    decoder_target_data[i, t:, target_token_index.get(' ')] = 1.

encoder_inputs = tf.keras.layers.Input(shape=(None, num_encoder_tokens))
encoder = tf.keras.layers.LSTM(latent_dim, return_state=True)
encoder_outputs, state_h, state_c = encoder(encoder_inputs)
encoder_states = [state_h, state_c]

encoder_input_datas = tf.keras.layers.Input(shape=(None,num_encoder_tokens))
encoder = tf.keras.layers.LSTM(latent_dim,return_state=True)
encoder_outputs, state_h, state_c = encoder(encoder_input_datas)
encoder_states = [state_h,state_c]

decoder_inputs = tf.keras.layers.Input(shape=(None,num_decoder_tokens))
decoder = tf.keras.layers.LSTM(latent_dim,return_sequences=True,return_state=True)
decoder_outputs,_,_ = decoder(decoder_inputs,initial_state=encoder_states)
decoder_dense = tf.keras.layers.Dense(num_decoder_tokens,activation='softmax')
decoder_outputs = decoder_dense(decoder_outputs)

model = tf.keras.models.Model([encoder_input_datas,decoder_inputs],decoder_outputs)

model.compile(optimizer='rmsprop', loss='categorical_crossentropy',metrics=['accuracy'])
model.fit([encoder_input_data,decoder_input_data],decoder_target_data,epochs=epochs)

**Other info / logs**
Full log:
https://pastebin.com/Yht9t6fZ

**Comment by ravikyram** [comment]
Looks like code is incomplete.In order to expedite the trouble-shooting process, please provide complete code snippet to reproduce the issue in our environment.It helps in localizing the issue faster. Thanks!

**Comment by DarshanDeshpande** [comment]
> Looks like code is incomplete.In order to expedite the trouble-shooting process, please provide complete code snippet to reproduce the issue in our environment.It helps in localizing the issue faster. Thanks!

https://github.com/DarshanDeshpande/Seq2Seq-Model
This is the link to the project along with the files used. I did some reading and found out a similar issue to this. The person fixed it by downgrading their Nvidia drivers. I tried that too but that doesn't seem to work either

**Comment by ymodak** [comment]
You may try gpu memory resources management by allowing gpu memory growth.  
To know more see https://www.tensorflow.org/guide/gpu#limiting_gpu_memory_growth
You can try Allowing GPU memory growth with:
```python
import tensorflow as tf
gpu = tf.config.experimental.list_physical_devices('GPU')
tf.config.experimental.set_memory_growth(gpu[0], True)
```

**Comment by DarshanDeshpande** [comment]
> Apologies for the delay in response. Is this still an issue?
> You may try gpu memory resources management by allowing gpu memory growth.
> To know more see https://www.tensorflow.org/guide/gpu#limiting_gpu_memory_growth
> You can try Allowing GPU memory growth with:
> 
> ```python
> import tensorflow as tf
> gpu = tf.config.experimental.list_physical_devices('GPU')
> tf.config.experimental.set_memory_growth(gpu[0], True)
> ```

I tried this but it didn't work out for me. There's no problem with the code as such because it ran fine on my colab notebook. Maybe it's just the GPU driver issue or something. Anyway thanks for the help. Closing this topic now

**Comment by sxooler** [comment]
@DarshanDeshpande: Were you able to find out what was the issue or not? I see you've closed the issue with statement it might be GPU driver issue. Not sure if it's true as I am working on Windows machine with different setup (TF 2.0.1, CUDA 10.1, cuDNN 7.6.5.32, GeForce RTX 2070 with Max-Q Design) and it works in some cases (especially with lower batch size) and crashes randomly. Any help would be appreciated.

**Comment by jensb1** [comment]
Same error here, TF 2.2.0rc3, CUDA10.1, cuDNN 7.6.5.32, GeForce RTX 2070 with Max-Q Design. 

I have tried 
```
gpu = tf.config.experimental.list_physical_devices('GPU')
tf.config.experimental.set_memory_growth(gpu[0], True)
```

Simple model: 

```
from tensorflow.keras import layers
from tensorflow.keras import models

def make_model():

    model = models.Sequential([
        layers.LSTM(300, batch_input_shape=(8, 4096, 25), return_sequences=True),
        layers.LSTM(100, return_sequences=True),  
        layers.Dense(1, activation='sigmoid')
    ])
    
    model.compile(
      optimizer="adam",
      loss='binary_crossentropy'
    )
    return model

```
